backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cgen/CBridgeGen.kt

```diff
@@ -820,6 +820,10 @@ private fun KotlinStubs.mapType(
                 ?: reportUnsupportedType("not a structure or too complex"))
     }
 
+    type.classOrNull?.isSubtypeOfClass(symbols.nativePointed) == true -> {
+        ByRefValuePassing(symbols.nativePtrType)
```

Comment:
Isn't `TrivialValuePassing(type, CTypes.voidPtr)` enough?

Reply:
We need to perform conversion in `bridgedToKotlin` to dereference incoming pointer.